var/spack/repos/builtin/packages/py-numpy-groupies/package.py

```diff
@@ -23,10 +23,11 @@ class PyNumpyGroupies(PythonPackage):
 
     license("BSD-2-Clause")
 
+    version("0.11.2", sha256="2fda978c4d28d2f1633a63972f425d0a7f2f12a75505d215b41b6de712e2ec4b")
     version("0.9.20", sha256="923a382d6bc6876384b58a9c0503b05b9d36a660f329695c2d33e4f93fcbbe3d")
 
-    depends_on("python@3.7:", type=("build", "run"))
-
+    depends_on("python@3.7:3.11", type=("build", "run"), when="@0.9.20")
```

Comment:
Better to have a when clause with upper limit, e.g. (unverified!):
```suggestion
    depends_on("python@3.7:3.11", type=("build", "run"), when="@:0.9")
```

Reply:
confirmed python@:3.11 is required for all versions older than 0.11.2